platform/security/certificate/src/main/java/org/codice/ddf/certificate/PEMManager.java

```diff
@@ -0,0 +1,142 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.openssl.PEMDecryptorProvider;
+import org.bouncycastle.openssl.PEMEncryptedKeyPair;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.openssl.jcajce.JcePEMDecryptorProviderBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+        /*+----------------------------------------------------------------------
+         ||
+         ||  Class: PEMManager
+         ||
+         ||        Purpose:  Collection of static methods that extracts objects from PEM files. Currently PEMKeyPair, PrivateKeys, and X509Certificate's.
+         ||
+         ||  Inherits From:  None
+         ||
+         ||     Interfaces:  None
+         ||
+         |+-----------------------------------------------------------------------
+         ||
+         ||      Constants:  None
+         ||
+         |+-----------------------------------------------------------------------
+         ||
+         ||  Class Methods:     X509Certificate     loadX509FromPEM(String)
+         ||                     PrivateKey          getPrivateKeyFromPEMFile(String, String)
+         ||
+         ||
+         ++-----------------------------------------------------------------------*/
+
+public class PEMManager {
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: getPrivateKeyFromPEMFile
+     * <p/>
+     * Purpose:  Given a PEM file that contains a PEMEncryptedKeyPair or PrivateKeyInfo object, this method extracts and returns the private key contained within.
+     * <p/>
+     * Pre-condition:  None
+     * <p/>
+     * Post-condition: The privateKey found in the PEM file is returned
+     *
+     * @param fileName the name/path of the PEM file to use
+     * @param password password used to decrypt the privateKey
+     * @return PrivateKey that was extracted from the PEM file
+     * -------------------------------------------------------------------
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(PEMManager.class);
+
+    public static PrivateKey getPrivateKeyFromPEMFile(String fileName, String password) {
+        KeyPair kp;
+        PrivateKey privKey = null;
+        CertificateManager.registerBCSecurityProvider();
+        try {
+            File privateKeyFile = new File(fileName);
+            PEMParser pemParser = new PEMParser(new FileReader(privateKeyFile));
```

Comment:
We should probably check that the file is readable and exists before reading it.